### PR TITLE
chore: Refactoring header actions on datasource page to support reusable queries on EE

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Refactoring header actions on datasource page to support reusable queries on EE

#### PR fixes following issue(s)
Fixes [#28568](https://github.com/appsmithorg/appsmith/issues/28568)

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced a new header action management system in the data source editor, allowing for dynamic action button generation based on user permissions and editor type.
  - Implemented a new editor type detection hook to improve editor experience.

- **Improvements**
  - Streamlined the data source editor interface by removing unused permissions checks and simplifying action button logic.
  - Enhanced the editor's routing system with additional constants for deprecated paths.

- **Refactor**
  - Refactored the `DSFormHeader` component to utilize the new `useHeaderActions` hook for a cleaner and more maintainable codebase.
  - Updated Redux state management to include a new action handler for resetting editor state.

- **Style**
  - Adjusted the layout of the `LeftPaneContainer` by removing the `min-width` property for a more responsive design.

- **Chores**
  - Cleaned up imports and unused variables across multiple files to improve code clarity and maintainability.

### CODE REVIEW

File: app/client/src/ce/hooks/datasourceEditorHooks.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    datasource,
+    isPluginAuthorized,
+    pluginType,
+    showReconnectButton = false,
+  }: {
+    datasource: Datasource | ApiDatasourceForm | undefined;
+    isPluginAuthorized: boolean;
+    pluginType: string;
+    showReconnectButton?: boolean;
+  },
```
Comment (ashit-rath) [inline app/client/src/ce/hooks/datasourceEditorHooks.tsx:41]: Can you create an `interface` out of it?

File: app/client/src/ce/hooks/datasourceEditorHooks.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        >
+          {createMessage(GENERATE_NEW_PAGE_BUTTON_TEXT)}
+        </Button>
+      ) : null;
+    return {
+      newActionButton,
+      generatePageButton,
+    };
+  }
+  return {};
```
Comment (ashit-rath) [inline app/client/src/ce/hooks/datasourceEditorHooks.tsx:113]: Can we add a new line above

File: app/client/src/ce/hooks/datasourceEditorHooks.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            e.stopPropagation();
+            e.preventDefault();
+            routeToGeneratePage();
+          }}
+          size="md"
+        >
+          {createMessage(GENERATE_NEW_PAGE_BUTTON_TEXT)}
+        </Button>
+      ) : null;
+    return {
```
Comment (ashit-rath) [inline app/client/src/ce/hooks/datasourceEditorHooks.tsx:108]: Add a new line before this

File: app/client/src/ce/hooks/index.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export interface EditorType {
+  [key: string]: string;
+}
+
+export const useEditorType = (path: string) => {
+  const basePath = matchPath(path, {
+    path: [BUILDER_VIEWER_PATH_PREFIX, BUILDER_BASE_PATH_DEPRECATED],
+  });
+  const editorType: EditorType = {
```
Comment (ashit-rath) [inline app/client/src/ce/hooks/index.ts:19]: Can we move this outside the hook?

File: app/client/src/ce/reducers/uiReducers/editorReducer.tsx [hunk 5]
```diff
@@ -42,6 +42,18 @@ export const initialState: EditorReduxState = {
 };
 
 export const handlers = {
+  [ReduxActionTypes.RESET_EDITOR_REQUEST]: (state: EditorReduxState) => {
+    return {
```
Comment (ashit-rath) [inline app/client/src/ce/reducers/uiReducers/editorReducer.tsx:46]: Do you think we should return in the initialState to reset everything?

Comment (ankitakinger) [reply app/client/src/ce/reducers/uiReducers/editorReducer.tsx:46]: I'm not sure if that will break anything, hence did this.

Comment (ashit-rath) [reply app/client/src/ce/reducers/uiReducers/editorReducer.tsx:46]: Okay makes sense

**Pull request merged**